spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -817,6 +848,22 @@ def measure(self):
                 atlas_data_dct[tract_id] = img_cur_copy.data
                 del img_cur
 
+        # iterate across lesions to get the largest lesion for the midsagittal slice interpolation
+        for lesion_label in label_lst:
+            im_lesion_data_cur = np.copy(im_lesion_data == lesion_label)
+            label_idx = self.measure_pd[self.measure_pd.label == lesion_label].index
+            self._measure_volume(im_lesion_data_cur, p_lst, label_idx)
+        # Get the index of the largest lesion
+        largest_lesion_idx = self.measure_pd.loc[pd.to_numeric(self.measure_pd['volume [mm3]']).idxmax()]['label']
+        printv(f'Largest lesion index: {largest_lesion_idx}', self.verbose, 'info')
+        # Get sagittal slices and compute weights to be used for the midsagittal slice interpolation
+        # NOTE: the interpolation will be done separately for each lesion in the for loop below
+        im_lesion_data_largest_lesion = np.copy(im_lesion_data == largest_lesion_idx)
+        self.midsagittal_slice_interpolation(im_lesion_data_largest_lesion)
+        # NOTE: the interpolated midsagittal slice is the same for all lesions
+        printv(f'Interpolated midsagittal slice (same across lesions) = '
+               f'{round(self.interpolated_midsagittal_slice, 2)}', self.verbose, 'info')
+
```

Comment:
See: Previous comment...

All of this extra logic could live in the same location as the old `get_midsagittal_slice` function (to keep the diff neat and tidy).

Reply:
done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4702/commits/11cdd6567853945cd189a02a38da4c5b1838fc45